### ISSUE TITLE
System order reporting and visualization

### ISSUE THREAD

**Issue by alice-i-cecile** [bevyengine/bevy#2137]
## What problem does this solve or what need does it fill?

The order in which systems can execute is critically important for understanding and debugging but this information is typically spread across an entire app.

## What solution would you like?

1. Provide a user-friendly method to log the schedule. This should be human-readable to both ease parsing and make it usable for vision-impaired users.
2. Create a graph visualization from this log.
3. Create an example showing how to use these features.

## What alternative(s) have you considered?

None.

## Additional context

This has been discussed informally repeatedly, but I somehow couldn't find an issue.

https://github.com/jakobhellermann/bevy_mod_debugdump is fantastic prior art.

Other more advanced functionality (e.g. filter by component / resource type, showing which execution order was actually used, filtering by stage or module) can be added on top of this once the initial PR is merged. Those should have separate issues for clarity.

**Comment by alice-i-cecile** [comment]
Ideally this should be integrated with #2025 as well.

**Comment by dimvoly** [comment]
What's the current method for debugging system execution order problems?

It seems the best way right now is to guess & check which is quite ineffective.

I've got code where I've put things into sets and either they don't run as expected or run out of order. Despite using .chain() and .run_if() etc.

Be nice to have something that could be dumped out to a text file, even if it's quite ugly.
